File: baekjoon-Algorithm/algorithm-middle/greedy-algorithm/N10162.java
import java.util.*;
import java.io.*;

public class N10162 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());

        int modA = T / 300;
        int remainA = T % 300;

        int modB = remainA / 60;
        int remainB = remainA % 60;

        int modC = remainB / 10;
        int remainC = remainB % 10;

        if (remainC != 0) {
            System.out.println(-1);
        } else {
            System.out.println(modA + " " + modB + " " + modC);
        }
    }
}

File: baekjoon-Algorithm/algorithm-middle/greedy-algorithm/N10162.py
import sys

input = sys.stdin.readline

T = int(input())

modA = T // 300
remainA = T % 300

modB = remainA // 60
remainB = remainA % 60

modC = remainB // 10
remainC = remainB % 10

if remainC != 0:
    print(-1)
else:
    print(modA, modB, modC, sep=" ")
